subsys/net/ip/net_context.c

```diff
@@ -816,6 +816,18 @@ NET_CONN_CB(tcp_established)
 				     sys_get_be32(NET_TCP_HDR(pkt)->ack));
 	}
 
+	if ((s32_t)(sys_get_be32(NET_TCP_HDR(pkt)->seq) - context->tcp->send_ack) < 0) {
```

Comment:
The check above is exactly coded to give correct answer even if seq numbers wrapped. But yes, ideally instead of that more or less complex check, there would be a semantically clearer macro/incline function call. Didn't have time to do that on weekend, so coded independently like above, will work on ZEP-2120 now.